app/actions/remote/post.ts

```diff
@@ -341,7 +331,8 @@ export const fetchPostsForUnreadChannels = async (serverUrl: string, channels: C
             promises.push(fetchPostsForChannel(serverUrl, channel.id));
         }
     }
-    await Promise.all(promises);
+    const responses = await Promise.all(promises);
+    return {data: responses};
```

Comment:
modified in another PR check this one https://github.com/mattermost/mattermost-mobile/pull/8306/files#diff-159b2de61799d5939de9341fba89b5310440deba6fe35f84025edb79a49b02e6R346 and you can change it to be the same, no need to await here just return the Promise.all

```suggestion
    return Promise.all(promises);
```